### PR TITLE
POL-617 - Deprecate authentication `type: "rightscale"` 

### PR DESCRIPTION
**Pull request by bryankaraffa**
### Description

This PR brings the required changes to standardize the policy Catalog between US and EU.

Changes required to deprecate `EU_Policies` branch:
 - Replaced auth_rs with auth_flexera (bb7d2d87f5b39ab9657b31aa709cea061c1df735)
 - Modify `sys_log` definition to prevent `rs_cm.audit_entry.create` outside Flexera NAM zone (cbfc00ab062717ddf8a0fdf3b7594414491d18bf)
 - Bumped template version and add CHANGELOG notes (dc1e3a3a835a4314a2a2a832307a3d5e9174de0a, f5b97af3923ce45e52203ede9526ff57b79a8ddc)
 -  Updated `Prerequisites` section in `README.md`(b7847fa758c5390ed28c56bb6417f27086e5ae14) with requirements for Flexera Credential

Other changes included in this branch:
 - Removed `sys_log` definition where unused (6a86d0b4380976a332558bb8c70616a63c8d6446)
 - Fixed missing `delete_unused_sql_databases` definition (c6c351d20bf86f3ab156d93b4eea3783912c1f75)
 - Replaced references `github.com/rightscale/policy_templates` and `github.com/flexera/policy_templates` with `github.com/flexera-public/policy_templates` (2f2da7af40d4fa43052a272f3ecf57710b316a39)
 - Disabled publishing *AzureAD Group Sync* policy (18f1b2aff5b4ffb026af33aa0422a9abd361cdeb)

### Issues Resolved

 - [POL-617](https://flexera.atlassian.net/browse/POL-617)

### Contribution Check List

- [x] All tests pass.
- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable
- [X] New functionality has been documented in CHANGELOG.MD
- [x] [Provider-Specific Credentials](https://docs.flexera.com/flexera/EN/Automation/ProviderCredentials.htm) page updated to include docs for creating Credential for `Flexera` [`provider=flexera`]

After this is approved/merged, pushing changes to production is mostly automated
- [ ] [Upload to S3](https://github.com/flexera-public/policy_templates/actions/workflows/upload-files.yaml) workflow runs after successful [Test Policies on `master` branch](https://github.com/flexera-public/policy_templates/actions/workflows/test-policies.yaml?query=branch%3Amaster). We can/should validate this happens as expected
- [ ] [Flexera NAM Policy Sync Applied Policy](https://app.flexera.com/orgs/5720/policy/projects/summary/applied-policies/61e9d33bd4d7da000198786f) will update and publish new versions.  We can/should validate this happens as expected.
- [ ] Update [Flexera EU Policy Sync Applied Policy](https://app.flexera.eu/orgs/30351/policy/projects/130351/applied-policies/62ab599e1e8db50001139468) branch parameter from `EU_Policies` to `master` branch.  Next policy run, the new versions will get published in Flexera EU. After updating value and running, we can/should validate the updates are as expected and match what was deployed to Flexera NAM

### CODE REVIEW

File: compliance/aws/iam_role_audit/CHANGELOG.md
Comment (bryankaraffa) [inline compliance/aws/iam_role_audit/CHANGELOG.md:5]:
```suggestion
- Removed unused `sys_log` definition
```

File: compliance/fnms/fnms_licenses_expiring/README.md
Comment (bryankaraffa) [inline compliance/fnms/fnms_licenses_expiring/README.md:8]: @cdwilhelm - Can you help me confirm what permissions are needed for these ITAM policies that use the `api.flexera.com/fnms/v1/orgs/....` endpoint please?  I assumed fnms_user or fnms_admin for these because they were not defined in the previous README

Comment (cdwilhelm) [reply compliance/fnms/fnms_licenses_expiring/README.md:10]: I'm not going to be much help here.  The FNMS policies were written by solution enginers.  I would suggest connecting with an SE who is familiar with FNMS/Itam.

**Review by taterdog590 — DISMISSED** [review]
LGTM

**Comment by bryankaraffa** [comment]
@cdwilhelm, @flexera-public/policy-developers -- I am proposing we release this Monday.  I need at least one formal approval on this PR before we can merge, and once we merge changes that will trigger automation to publish new versions.  Curt will also make a post in the Flexera Community to raise awareness of this change and new requirement.

I used a script to deploy the policy templates which had the `type: rightscale` change and they all run as expected using `auth_flexera` credential instead of rightscale credential.
image

**Comment by bryankaraffa** [comment]
All merge conflicts handled for what's in mater right now.  I am hoping we can hold off on any other major commits to master before we can merge this (currently planned for Monday when @cdwilhelm gets back.
CC: @nia-vf1 @taterdog590 @XOmniverse @lawrethao

**Review by nia-vf1 — COMMENTED** [review]
Sorry this is going to be a long comment - Just looked through all the files. Noticed there are some policies that still have the credential has "auth_rs" instead of "auth_flexera"

I have made some assumptions:

- Some policies with auth_rs are deprecated so I assume we can ignore those here
- I also noticed that the ITAM (fnms) and SaaS Manager (saas/fsm) policies are using "auth_flexeraone" and "fsm_auth" respectively, but I am assuming that these not being updated is intentional?

However, the policies below I believe should be updated to use "auth_flexera" but may have been missed are:

-	Compliance/policy_update_notification
-	Cost/aws/reserved_instances/recommendations
-	_Cost/Aws/savings_plan/utilization_
-	_Cost/Aws/savings_realized_
-	_Cost/Azure/savings_realized_
-	_Cost/forecasting/commitment_forecast_
-	_Cost/forecasting/straight_line_forecast/linear_regression_
-	_Cost/forecasting/straight_line_forecast/simple_
-	Cost/scheduled_report_markupsdowns
-	Cost/scheduled_reports_with_estimates
-	msp/cost/master_org_cost_policy
-	msp/cost/master_org_cost_policy_currency
-	operational/applied_policy_error_notification
-	_operational/aws/total_instance_hours_
-	_operational/aws/total_instance_hours_forecast_
-	_operational/aws/total_instance_vcpus_
-	_operational/aws/total_instance_vcpus_forecast_
-	operational/azure/azuread_group_sync
-	operational/bill_processing_errors_notification

Many of these policies (in italics) I have created recently, so that could be why they got missed. I happy to open another pull request to address these. The other policies in this list I haven't really worked with so not even sure how often they are used - if these were intentionally not updated let me know @bryankaraffa 

TLDR: Happy to approve this but will need another pull request to update the credentials for the few policies listed above. The ones in italics I can update, however I would like confirmation on the others as them not being updated could have been intentional.

**Review by nia-vf1 — APPROVED** [review]
LGTM

**Pull request merged**
